app/models/concerns/profile_policy_association.rb

```diff
@@ -5,43 +5,18 @@ module ProfilePolicyAssociation
   extend ActiveSupport::Concern
 
   included do
-    after_destroy :destroy_policy_test_results
+    after_destroy :destroy_empty_policy
 
-    def policy
-      return self unless external
-
-      Profile.includes(:benchmark)
-             .older_than(created_at)
-             .find_by(account: account_id, external: false, ref_id: ref_id,
-                      benchmarks: { ref_id: benchmark.ref_id })
-    end
-
-    def policy_profiles
-      return Profile.none if account_id.nil?
-
-      Profile.includes(:benchmark)
-             .where(account: account_id, ref_id: ref_id,
-                    benchmarks: { ref_id: benchmark.ref_id })
-    end
```

Comment:
These weird methods that pointed to other profiles now point to policies in the DB.